inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/KafkaMultiTopicsFetcher.java

```diff
@@ -0,0 +1,388 @@
+        this.setStopConsume(true);
+
+        // update
+        this.onlineTopics = newTopics.stream().collect(Collectors.toMap(InLongTopic::getTopic, t -> t));
+        InLongTopic topic = onlineTopics.values().stream().findFirst().get();
+        this.seeker = SeekerFactory.createKafkaSeeker(consumer, topic);
+        this.listener = new AckOffsetOnRebalance(topic.getInLongCluster().getClusterId(), seeker,
+                commitOffsetMap, ackOffsetMap);
+        Optional.ofNullable(interceptor).ifPresent(i -> i.configure(topic));
+
+        // subscribe new
+        consumer.subscribe(onlineTopics.keySet(), listener);
+
+        // resume
+        this.setStopConsume(false);
+        return true;
+    }
+
+    private void prepareCommit() {
+        ackOffsetMap.forEach((topicPartition, tpOffsetMap) -> {
+            synchronized (tpOffsetMap) {
+                // get the remove list
+                List<Long> removeOffsets = new ArrayList<>();
+                long commitOffset = -1;
+                for (Long ackOffset : tpOffsetMap.keySet()) {
+                    if (!tpOffsetMap.get(ackOffset)) {
+                        break;
+                    }
+                    removeOffsets.add(ackOffset);
+                    commitOffset = ackOffset;
+                }
+                // the first haven't ack, do nothing
+                if (commitOffset == -1) {
+                    return;
+                }
+
+                // remove offset and commit offset
+                removeOffsets.forEach(tpOffsetMap::remove);
+                commitOffsetMap.put(topicPartition, new OffsetAndMetadata(commitOffset));
+            }
+        });
+    }
+
+    public class Fetcher implements Runnable {
+
+        private void commitKafkaOffset() {
+            prepareCommit();
+            if (consumer != null) {
+                try {
+                    consumer.commitAsync(commitOffsetMap, null);
+                    commitOffsetMap.clear();
+                } catch (Exception e) {
+                    LOGGER.error("commit kafka offset failed: {}", e.getMessage(), e);
+                }
+            }
+        }
+
+        /**
+         * put the received msg to onFinished method
+         *
+         * @param messageRecords {@link List < MessageRecord >}
+         */
+        private void handleAndCallbackMsg(List<MessageRecord> messageRecords) {
+            long start = System.currentTimeMillis();
+            try {
+                context.getDefaultStateCounter().addCallbackTimes(1);
+                context.getConfig().getCallback().onFinishedBatch(messageRecords);
+                context.getDefaultStateCounter()
+                        .addCallbackTimeCost(System.currentTimeMillis() - start)
+                        .addCallbackDoneTimes(1);
+            } catch (Exception e) {
+                context.getDefaultStateCounter().addCallbackErrorTimes(1);
+                LOGGER.error("failed to callback: {}", e.getMessage(), e);
```

Comment:
`LOGGER.error("failed to callback: ", e);`

Reply:
fixed, thx